### PR TITLE
Add methods for creating and sending requests

### PR DESCRIPTION
**Pull request by shadowhand**
- add `getAuthenticatedRequest` method to get a request builder
- add `getResponse` method to send requests and parse the response
- use new methods for handling user detail calls

Fixes thephpleague/oauth2-client#268

### CODE REVIEW

File: src/Provider/AbstractProvider.php [hunk 1]
```diff
@@ -363,32 +413,9 @@ protected function fetchUserDetails(AccessToken $token)
...
-
-            $httpResponse = $client->get($url, $headers);
-
-            $response = (string) $httpResponse->getBody();
-        } catch (HttpAdapterException $e) {
-            // @codeCoverageIgnoreStart
-            $response = (string) $e->getResponse()->getBody();
-            // @codeCoverageIgnoreEnd
-        }
+        $request = $this->getAuthenticatedRequest(Request::METHOD_GET, $url, $token);
```
Comment (shadowhand) [inline src/Provider/AbstractProvider.php]: @ramsey switching to use `getAuthenticatedRequest` and `getResponse` internally for user details.

**Comment by shadowhand** [comment]
In the future, it might be worthwhile to add a `getBaseUrl` method to providers that returns the root URL of the API, to allow partial path support when calling `getAuthenticatedRequest`.

File: src/Provider/AbstractProvider.php (hunk 1 above)
Comment (ramsey) [inline src/Provider/AbstractProvider.php]: Awesome!

**Comment by ramsey** [comment]
Yep. That sounds good to me.

**Pull request closed without merging**